src/components/Menu/menuItemRules.ts

```diff
@@ -42,7 +42,7 @@ export default {
       cursor: 'pointer',
       display: 'block',
       ...(shape === 'pills' && {
-        margin: `0 ${pxToRem(8)} 0 0`,
+        ...(vertical ? { margin: `0 0 ${pxToRem(5)} 0` } : { margin: `0 ${pxToRem(8)} 0 0` }),
```

Comment:
Maybe we should make the margin consistent for the pills menu for the vertical and horizontal type, both to be 5px or 8 px whatever you think looks better.